pkg/apiOracle/valueOracle.go

```diff
@@ -76,18 +78,22 @@ func writeOutHistory() {
 	psrSavedDataTmp := psrSavedData + ".tmp"
 	err = ioutil.WriteFile(psrSavedDataTmp, data, 0644)
 	if err != nil {
-		logger.Error("failed to write out PSR values to %s: %s", psrSavedDataTmp, err.Error())
+		level.Error(logger).Log(
+			"msg", "failed to write out PSR values",
+			"psrSavedDataTmp", psrSavedDataTmp,
+			"err", err.Error(),
```

Comment:
Here and few other places

```suggestion
			"err", err,
```